tests/unit/test_import.py

```diff
@@ -5,21 +5,33 @@
 import pytest
 
 
-@pytest.mark.unit
-def test_multiprocessing_init_non_win32(monkeypatch):
-    """Test multiprocessing init on non-Windows platforms"""
-    monkeypatch.setattr(sys, "platform", "linux")
-    with patch("multiprocessing.set_start_method") as mock_set_start_method:
-        del sys.modules["pybop"]
-        importlib.import_module("pybop")
-        mock_set_start_method.assert_called_once_with("fork")
+class TestImport:
+    @pytest.mark.unit
+    def test_multiprocessing_init_non_win32(self, monkeypatch):
+        """Test multiprocessing init on non-Windows platforms"""
+        monkeypatch.setattr(sys, "platform", "linux")
+        # Unload pybop and its sub-modules
+        self.unload_pybop()
+        with patch("multiprocessing.set_start_method") as mock_set_start_method:
+            importlib.import_module("pybop")
+            mock_set_start_method.assert_called_once_with("fork")
 
+    @pytest.mark.unit
+    def test_multiprocessing_init_win32(self, monkeypatch):
+        """Test multiprocessing init on Windows"""
+        monkeypatch.setattr(sys, "platform", "win32")
+        self.unload_pybop()
+        with patch("multiprocessing.set_start_method") as mock_set_start_method:
+            importlib.import_module("pybop")
+            mock_set_start_method.assert_called_once_with("spawn")
 
-@pytest.mark.unit
-def test_multiprocessing_init_win32(monkeypatch):
-    """Test multiprocessing init on Windows"""
-    monkeypatch.setattr(sys, "platform", "win32")
-    del sys.modules["pybop"]
-    with patch("multiprocessing.set_start_method") as mock_set_start_method:
-        importlib.import_module("pybop")
-        mock_set_start_method.assert_called_once_with("spawn")
+    def unload_pybop(self):
+        """
+        Unload pybop and its sub-modules. Credit PyBaMM team:
+        https://github.com/pybamm-team/PyBaMM/blob/develop/tests/unit/test_util.py
```

Comment:
could you change this to a permalink just in case the tests get moved or change later on and break here – we are having students from GSoC this year to improve PyBaMM's testing infrastructure, so this is in anticipation for that

```suggestion
        Unload pybop and its sub-modules. Credit PyBaMM team:
        https://github.com/pybamm-team/PyBaMM/blob/90c1c357a97dfd5c8c6a9092a70dddf0dac978db/tests/unit/test_util.py